orc8r/cloud/go/services/directoryd/client_api.go

```diff
@@ -125,6 +125,47 @@ func MapSessionIDsToIMSIs(networkID string, sessionIDToIMSI map[string]string) e
 	return nil
 }
 
+// MapHWIDToDirectoryRecordIDs maps {hwid -> directory record IDs}. If existing
+// record IDs already exist for a given hwid, new record IDs will be appended
+// to the existing state.
+func MapHWIDToDirectoryRecordIDs(networkID string, hwidToDirectoryRecordIDs map[string][]string) error {
+	client, err := getDirectorydClient()
+	if err != nil {
+		return errors.Wrap(err, "failed to get directoryd client")
+	}
+	hwidToDirectoryRecordIDProto := map[string]*protos.DirectoryRecordIDs{}
```

Comment:
Consider making explicit helper functions for these conversions between proto and native types